lib/rubygems/command.rb

```diff
@@ -152,15 +152,22 @@ def execute
   #--
   # TODO: replace +domain+ with a parameter to suppress suggestions
 
-  def show_lookup_failure(gem_name, version, errors, domain)
+  def show_lookup_failure(gem_name, version, errors, domain, required_by = nil)
+    gem = "'#{gem_name}' (#{version})"
+
```

Comment:
`msg` should be getting initialized here instead and then getting appended to. IMO this is cleaner as you're only creating `msg` once.

example: 

```
msg = StringIO.new

if errors and !errors.empty?
   msg << "Could not find a valid gem #{gem}, here is why:\n".dup
end
```

Reply:
Thanks, updated!